silq/meta_instruments/layout.py

```diff
@@ -1587,28 +1656,80 @@ def save_traces(self,
+            trace_file.attrs['final_loop_indices'] = active_measurement.loop_indices
 
-        if name in self.trace_files:  # Use existing trace file
-            trace_file = self.trace_files[name]
-        else:  # Create new trace file
-            trace_file = self.initialize_trace_file(name=name, folder=folder)
-            self.trace_files[name] = trace_file
+        return trace_file
 
-        traces = self.acquisition_interface.traces
-        for channel in channels:
-            # Get corresponding acquisition output channel name (chA etc.)
-            ch = next(ch_pair[0] for ch_pair in self.acquisition_channels()
-                      if ch_pair[1] == channel)
-            trace_file['traces'][channel][active_measurement.loop_indices] = traces[ch]
-        trace_file.attrs['final_loop_indices'] = active_measurement.loop_indices
+    def plot_traces(self, channel_filter: str = None, **plot_kwargs):
+        """Plot acquisition traces for acquisition channels
 
-        return trace_file
+        Args:
+            channel_filter: Optional filter for channel names.
+                If passed, only channels containing this string are plotted
+            **plot_kwargs: Optional kwargs passed when adding traces
+
+        Returns:
+            MatPlot object
+        """
+        assert channel_filter is None or isinstance(channel_filter, str)
+
+        traces_channels = self.acquisition_interface.traces
+        # TODO add check if no traces have been acquired
+
+        # Map traces to the channel labels
+        channel_mappings = dict(self.acquisition_channels())
+        traces = {channel_mappings[ch]: trace_arr
+                  for ch, trace_arr in traces_channels.items()
+                 }
+        trace_shape = next(iter(traces.values())).shape
+
+        # Optionally filter outputs
+        if channel_filter:
+            if isinstance(channel_filter, str):
+                channel_filter = [channel_filter]
+            traces = {key: val for key, val in traces.items()
+                     if any(name in key for name in channel_filter)}
+
+        t_list = np.arange(trace_shape[1]) / self.sample_rate()
+        if not self.acquisition_interface.capture_full_trace():
+            t_list += min(self.acquisition_interface.pulse_sequence.t_start_list)
+        t_list *= 1e3  # Convert to ms
+        sample_list = np.arange(trace_shape[0], dtype=float)
+
+        plot = MatPlot(subplots=len(traces))
+        for k, (ax, (channel, traces_arr)) in enumerate(zip(plot, traces.items())):
+                if traces_arr.shape[0] == 1:
+                    ax.add(traces_arr[0], x=t_list, **plot_kwargs)
+                    ax.set_ylabel('Amplitude (V)')
+                else:
+                    ax.add(traces_arr, x=t_list, y=sample_list, **plot_kwargs)
+                ax.set_ylabel('Sample number')
+                ax.set_xlabel('Time (ms)')
+                ax.set_title(channel)
+
```

Comment:
Could we make `DataArray`s here so the units are handled automatically in MatPlot?

Reply:
Good idea! I added it and looks good